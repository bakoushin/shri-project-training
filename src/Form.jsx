import React from 'react';
import {Link} from 'react-router-dom';

const Form = () => (
  <div>
    <h1>Hey! </h1>
    <p>Blah blah blah</p>
    <Link to="/card">card</Link>
  </div>
);

export default Form;
