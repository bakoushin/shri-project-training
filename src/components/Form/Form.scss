$bg: peru;

.Form {
  background: peachpuff;
  display: flex;
  &:hover {
    background: $bg;
  }
}
