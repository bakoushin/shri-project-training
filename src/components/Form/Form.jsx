import React, {Component} from 'react';
import {Link} from 'react-router-dom';
import Status from '../Status/Status';
import './Form.scss';

class Form extends Component {
  constructor(props) {
    super(props);
    this.state = {
      inputData: ''
    };
    this.inputChangeHandler = this.inputChangeHandler.bind(this);
  }
  inputChangeHandler(event) {
    this.setState({inputData: event.target.value});
  }
  render() {
    return (
      <div className="Form">
        <h1>Hey! </h1>
        <p>Blah blah blah</p>
        <input type="text" value={this.state.inputData} onChange={this.inputChangeHandler} />
        <p className="Form-Data">{this.state.inputData}</p>
        <Link to="/card">card</Link>
        <Status />
      </div>
    );
  }
}

export default Form;
