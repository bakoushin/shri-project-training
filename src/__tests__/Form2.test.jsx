import React from 'react';
import {shallow} from 'enzyme';
import Form from '../components/Form/Form';

test('Form rendres correctly', () => {
  const component = shallow(<Form />);
  expect(component).toMatchSnapshot();
});
