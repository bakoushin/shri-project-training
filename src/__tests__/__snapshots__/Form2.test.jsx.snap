// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form rendres correctly 1`] = `
<div
  className="Form"
>
  <h1>
    Hey! 
  </h1>
  <p>
    Blah blah blah
  </p>
  <BrowserRouter>
    <Link
      replace={false}
      to="/card"
    >
      card
    </Link>
  </BrowserRouter>
</div>
`;
